h/schemas/api/group.py

```diff
@@ -50,6 +50,10 @@ def validate(self, data):
         :rtype: dict
 
         """
+        # Strip leading and trailing whitespace from group names.
+        if isinstance(data.get("name"), str):
+            data["name"] = data["name"].strip()
```

Comment:
Just trim leading and trailing whitespace from the group name before validating it. So this means that non only are whitespace-only gruop names invalid, but also names that're less than the minimum length after trimming leading and trailing whitespace. Also if a name is still long enough after trimming whitespace, the name that gets saved will be the whitespace-trimmed version not the name that was actually given.

The alternative would be to raise a validation error if the name starts or ends with whitespace.